### ISSUE TITLE
[Possible bug] VoodooInput v1.0.8 - Not working gestures

### ISSUE THREAD

**Issue by dreamwhite** [acidanthera/bugtracker#1206]
Hi there,
I apologize if this isn't the right place where to ask but today I've noticed that if I use VoodooInput v.1.0.8 bundled with VoodooPS2 v2.1.7, my MSFT0001 trackpad gestures aren't recognized at all but pointer and two-finger scroll works.
Otherwise, if I use VoodooInput bundled with VoodooI2C v2.4.3, the gestures works. 

Below I attach my two EFI folders (with not-working gestures and with working ones):

[Working gestures](https://github.com/dreamwhite/dell-inspiron-5370-hackintosh/releases/download/0.6.2/EFI.zip)
[Not working gestures](https://github.com/acidanthera/bugtracker/files/5348620/dell-inspiron-5370-hackintosh-0.6.2.zip)

Please let me know if you need any type of logs and gently tell me how to obtain them.
Many thanks

**Comment by usr-sse2** [comment]
Do kexts load at all?
`kextstat | grep oodoo`

**Comment by dreamwhite** [comment]
> Do kexts load at all?
> `kextstat | grep oodoo`

Thank you so much for your reply.
Here you are the kextstat output with the working version of VoodooInput: 
[kextstat_working.txt](https://github.com/acidanthera/bugtracker/files/5350802/kextstat_working.txt)

I'll edit this comment and send you again the kextstat while the trackpad isn't working

**Comment by vit9696** [comment]
TL;DR You have touchpad support implemented by VoodooI2C (`com.alexandred.VoodooI2CHID`) and not by VoodooPS2. This means that your VoodooInput kext should come from VoodooI2C and not from VoodooPS2.

In more detail, we do not keep ABI (binary) compatibility between all VoodooInput releases, but only API compatibility (and in fact not always that even). That is why it is normal that a newer VoodooInput kext may not work with the driver expecting older API and ABI (VoodooI2CHID kext in your kext). The particular change that requires edits in the VoodooI2C code was introduced in VoodooInput 1.0.7. The support for it was added in VoodooI2C soon after, but I believe VoodooI2C binaries with this additions appeared only now in [2.5.1](https://github.com/VoodooI2C/VoodooI2C/releases).

Closing this.

**Comment by dreamwhite** [comment]
> TL;DR You have touchpad support implemented by VoodooI2C (`com.alexandred.VoodooI2CHID`) and not by VoodooPS2. This means that your VoodooInput kext should come from VoodooI2C and not from VoodooPS2.
> 
> In more detail, we do not keep ABI (binary) compatibility between all VoodooInput releases, but only API compatibility (and in fact not always that even). That is why it is normal that a newer VoodooInput kext may not work with the driver expecting older API and ABI (VoodooI2CHID kext in your kext). The particular change that requires edits in the VoodooI2C code was introduced in VoodooInput 1.0.7. The support for it was added in VoodooI2C soon after, but I believe VoodooI2C binaries with this additions appeared only now in [2.5.1](https://github.com/VoodooI2C/VoodooI2C/releases).
> 
> Closing this.

Thank you so much. I managed to solve by using VoodooInput by VoodooI2C and editing SSDT-GPI0 as per VoodooI2C issue [#388](https://github.com/VoodooI2C/VoodooI2C/issues/388)
